sdks/java/extensions/google-cloud-platform-core/src/main/java/org/apache/beam/sdk/extensions/gcp/util/Transport.java

```diff
@@ -125,7 +126,17 @@ private static HttpRequestInitializer httpRequestInitializerFromOptions(GcsOptio
 
     // Set custom audit info in request headers
     String jobName = Optional.ofNullable(options.getJobName()).orElse("UNKNOWN");
-    retryHttpRequestInitializer.setHttpHeaders(ImmutableMap.of("x-goog-custom-audit-job", jobName));
+
+    ImmutableMap.Builder<String, String> builder =
+        new ImmutableMap.Builder<String, String>().put("x-goog-custom-audit-job", jobName);
```

Comment:
Nit: Should we make `CUSTOM_AUDIT_JOB_ENTRY_KEY` public so that it can be referenced from here?